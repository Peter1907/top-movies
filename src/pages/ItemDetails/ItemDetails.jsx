import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import { getDetails, getStoredDetails } from '../../redux/details/details';
import s from './ItemDetails.module.css';

export default function ItemDetails() {
  const star = ('../assets/star.png');
  const saveIcon = ('../assets/add.png');
  const dispatch = useDispatch();
  const { id } = useParams();

  const Data = useSelector((state) => state.details);
  const storedData = localStorage.getItem(`D_${id}`);

  useEffect(() => {
    (storedData ? dispatch(getStoredDetails(id)) : dispatch(getDetails(id)));
    window.scrollTo(0, 0);
  }, [dispatch, id, storedData]);

  return (
    <div className={s.detailsContainer}>
      <div className={s.posterContainer}>
        <img className={s.saveTag} src={saveIcon} alt="#" />
        <img className={s.poster} src={Data.image} alt="poster" />
      </div>
      <div className={s.name}>
        <div className={s.header}>{Data.title}</div>
        {(!Data.tvSeriesInfo) && <div className={s.subHeader}>
          {Data.type} &spades; {Data.year} &spades; {`${Data.contentRating} `}
          &spades; {` ${Data.runtimeStr}`}
        </div>}
        {(Data.tvSeriesInfo) && <div className={s.subHeader}>
          {Data.type} &spades; {Data.year} &spades; {Data.contentRating}
        </div>}
      </div>
      <div className={s.rating}>
        <p className={s.rTitle}>Rating</p>
        <div>
          <img src={star} className={s.star} alt="star shaped icon" />
          <div className={s.rate}>
            <p className={s.score}><strong>{Data.imDbRating}</strong>/10</p><br />
            <p className={s.votes}>({Data.imDbRatingVotes})</p>
          </div>
        </div>
      </div>
      {(Data.genreList)
        && <ul className={s.genres}>
          {Data.genreList.map((genre, id) => (
            <li className={s.genre} key={id}>{genre.value}</li>
          ))}
        </ul>}
      <p className={s.plot}>{Data.plot}</p>
      <hr />
      <div className={s.creators}>
        <h3 className={s.cTitle}>Creators:</h3>
        {(Data.tvSeriesInfo) && <p>{Data.tvSeriesInfo.creators}</p>}
        {(Data.writerList) && <p>{Data.writers}</p>}
      </div>
      <hr />
      <div className={s.stars}>
        <h3 className={s.sTitle}>Stars:</h3>
        <p>{Data.stars}</p>
      </div>
      <hr />
      <div className={s.awards}>
        <h3 className={s.awardsT}>Awards:</h3>
        <p>{(Data.awards === '') ? 'none yet' : Data.awards}</p>
      </div>
      <hr />
      <div className={s.releaseDate}>
        <h3 className={s.releaseDateT}>Release Date:</h3>
        <p>{Data.releaseDate}</p>
      </div>
      <hr />
      <div className={s.companies}>
        <h3 className={s.companiesT}>Companies:</h3>
        <p>{Data.companies}</p>
      </div>
      <hr />
      <div className={s.countries}>
        <h3 className={s.countriesT}>Countries:</h3>
        <p>{Data.countries}</p>
      </div>
      <div className={s.cast}>
        <div className={s.castHead}>
          <h2>Top Cast</h2>
          <p className={s.arrow}></p>
        </div>
        {(Data.actorList) && <div className={s.castList}>
          {Data.actorList.map((actor, id) => (
            <div key={id} className={s.actor}>
              <div className={s.imgContainer} style={{
                backgroundImage: `url(${actor.image})`,
                backgroundSize: 'cover'
              }}>
              </div>
              <div className={s.actorInfo}>
                <h3>{actor.name}</h3>
                <p>{actor.asCharacter}</p>
              </div>
            </div>
          ))}
        </div>}
      </div>
    </div>
  );
}
