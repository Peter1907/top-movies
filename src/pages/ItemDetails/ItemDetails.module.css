.detailsContainer {
  background-color: #ec4c8a;
  color: white;
  min-height: calc(100vh - 8rem);
  padding: 5rem 0 3rem;
  text-align: center;
  display: flex;
  flex-direction: column;
  gap: 1rem;
}

.posterContainer {
  position: relative;
  margin: auto;
}

.poster {
  max-height: 80vh;
  max-width: 20rem;
}

.thumbnailContainer {
  width: 70%;
  position: relative;
  background-color: black;
  display: flex;
  overflow: hidden;
}

.saveTag {
  width: 3.5rem;
  position: absolute;
  top: 0;
  left: 1rem;
  opacity: 0.6;
  cursor: pointer;
  transition: all 250ms ease;
}

.saveTag:hover {
  opacity: 1;
  box-shadow: inset 0 0 0 80px #ffd369;
}

.header {
  font-size: 36px;
  font-family: 'Gill Sans', sans-serif;
  font-weight: 600;
  color: #ffd369;
}

.subHeader {
  color: white;
  margin-left: 1rem;
}

.rating {
  justify-self: stretch;
  display: flex;
  flex-direction: column;
  align-items: center;
  font-size: 22px;
  font-family: 'Gill Sans', sans-serif;
  margin: 1rem 0;
  background-color: #da457e;
  padding: 0.5rem;
}

.rTitle {
  color: #ffd369;
  font-weight: 600;
  letter-spacing: 1px;
  font-family: 'Gill Sans', sans-serif;
}

.score {
  font-size: 26px;
}

.votes {
  font-size: 14px;
  color: white;
  font-weight: 300;
}

.rating p {
  margin: 0;
}

.rating div {
  display: flex;
  align-items: center;
  gap: 0.5rem;
  text-align: center;
  margin: 0.5rem 0.5rem 0 0;
}

.rate {
  display: flex;
  flex-direction: column;
}

.star {
  width: 40px;
}

.genres {
  list-style-type: none;
  margin: 0 0 0.7rem;
  display: flex;
  justify-content: center;
  gap: 0.5rem;
  font-size: 26px;
  font-weight: 600;
  padding-left: 0;
}

.genre {
  border: 1px solid #ffd369;
  border-radius: 20px;
  padding: 0 10px;
  font-size: 16px;
}

.creators,
.stars,
.awards,
.releaseDate,
.companies,
.countries {
  display: flex;
  gap: 0.5rem;
  padding: 10px;
  align-items: baseline;
  justify-content: center;
}

.creators p,
.stars p,
.awards p,
.releaseDate p,
.companies p,
.countries p {
  margin: 1rem 0;
  line-height: 1.3rem;
  font-size: 18px;
  font-weight: 300;
}

.plot {
  margin: 1rem 0;
  line-height: 1.3rem;
  padding: 0 10px;
}

.cTitle,
.sTitle,
.awardsT,
.releaseDateT,
.companiesT,
.countriesT {
  margin: 0;
  font-weight: 600;
  color: #ffd369;
}

hr {
  width: 96%;
  color: white;
}

.cast {
  display: flex;
  flex-direction: column;
  gap: 1rem;
  margin: 5rem auto 2rem;
  padding: 0.5rem;
}

.castHead {
  display: flex;
  align-items: baseline;
  gap: 0.5rem;
  line-height: 4rem;
  margin: 0 auto 1rem;
}

.castHead h2 {
  font-family: 'Gill Sans', sans-serif;
  margin: 0;
  font-size: 42px;
  font-weight: 600;
}

.castList {
  display: flex;
  flex-wrap: wrap;
  margin: 0 auto;
  padding: 1rem;
  row-gap: 2rem;
  column-gap: 2.5rem;
  justify-content: center;
}

.actor {
  display: flex;
  flex-shrink: 0;
  gap: 0.5rem;
  width: 20rem;
  align-items: center;
  background: linear-gradient(rgba(0, 0, 0, 0.199), #2228314b);
  border-radius: 45%;
}

.actorInfo h3 {
  font-family: 'Gill Sans', sans-serif;
  margin: 0;
  font-size: 22px;
  letter-spacing: 0.5px;
}

.actorInfo p {
  margin: 0;
  padding-left: 7px;
}

.imgContainer {
  width: 7rem;
  height: 7rem;
  flex-shrink: 0;
  border-radius: 50%;
}

.arrow {
  border-top: 5px solid #fff;
  border-right: 5px solid #fff;
  height: 1rem;
  width: 1rem;
  transform: rotate(45deg);
  align-self: flex-end;
  margin: 0 0 1.4rem 0.5rem;
}
