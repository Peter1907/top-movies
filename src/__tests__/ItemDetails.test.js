import renderer from 'react-test-renderer';
import ItemDetails from '../_mocks_/ItemDetails';

it('renders correctly', () => {
  const tree = renderer
    .create(<ItemDetails />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
