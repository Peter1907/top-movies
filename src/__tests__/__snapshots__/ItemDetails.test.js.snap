// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="detailsContainer"
>
  <div
    className="posterContainer"
  >
    <img
      alt="#"
      className="saveTag"
      src="../assets/add.png"
    />
    <img
      alt="poster"
      className="poster"
      src="https://m.media-amazon.com/images/M/MV5BMDFkYTc0MGEtZmNhMC00ZDIzLWFmNTEtODM1ZmRlYWMwMWFmXkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_Ratio0.6762_AL_.jpg"
    />
  </div>
  <div
    className="name"
  >
    <div
      className="header"
    >
      The Shawshank Redemption
    </div>
    <div
      className="subHeader"
    >
      Movie
       ♠ 
      1994
       ♠ 
      R 
      ♠ 
       2h 22min
    </div>
  </div>
  <div
    className="rating"
  >
    <p
      className="rTitle"
    >
      Rating
    </p>
    <div>
      <img
        alt="star shaped icon"
        className="star"
        src="../assets/star.png"
      />
      <div
        className="rate"
      >
        <p
          className="score"
        >
          <strong>
            9.3
          </strong>
          /10
        </p>
        <br />
        <p
          className="votes"
        >
          (
          2652191
          )
        </p>
      </div>
    </div>
  </div>
  <ul
    className="genres"
  >
    <li
      className="genre"
    >
      Drama
    </li>
  </ul>
  <p
    className="plot"
  >
    Two imprisoned men bond over a number of years, finding solace and eventual redemption through acts of common decency.
  </p>
  <hr />
  <div
    className="creators"
  >
    <h3
      className="cTitle"
    >
      Creators:
    </h3>
    <p>
      Stephen King, Frank Darabont
    </p>
  </div>
  <hr />
  <div
    className="stars"
  >
    <h3
      className="sTitle"
    >
      Stars:
    </h3>
    <p>
      Tim Robbins, Morgan Freeman, Bob Gunton
    </p>
  </div>
  <hr />
  <div
    className="awards"
  >
    <h3
      className="awardsT"
    >
      Awards:
    </h3>
    <p>
      Top rated movie #1 | Nominated for 7 Oscars, 21 wins & 43 nominations total
    </p>
  </div>
  <hr />
  <div
    className="releaseDate"
  >
    <h3
      className="releaseDateT"
    >
      Release Date:
    </h3>
    <p>
      1994-10-14
    </p>
  </div>
  <hr />
  <div
    className="companies"
  >
    <h3
      className="companiesT"
    >
      Companies:
    </h3>
    <p>
      Castle Rock Entertainment
    </p>
  </div>
  <hr />
  <div
    className="countries"
  >
    <h3
      className="countriesT"
    >
      Countries:
    </h3>
    <p>
      USA
    </p>
  </div>
  <div
    className="cast"
  >
    <div
      className="castHead"
    >
      <h2>
        Top Cast
      </h2>
      <p
        className="arrow"
      />
    </div>
    <div
      className="castList"
    >
      <div
        className="actor"
      >
        <div
          className="imgContainer"
          style={
            Object {
              "backgroundImage": "url(https://m.media-amazon.com/images/M/MV5BMTI1OTYxNzAxOF5BMl5BanBnXkFtZTYwNTE5ODI4._V1_Ratio1.0000_AL_.jpg)",
              "backgroundSize": "cover",
            }
          }
        />
        <div
          className="actorInfo"
        >
          <h3>
            Tim Robbins
          </h3>
          <p>
            Andy Dufresne
          </p>
        </div>
      </div>
      <div
        className="actor"
      >
        <div
          className="imgContainer"
          style={
            Object {
              "backgroundImage": "url(https://m.media-amazon.com/images/M/MV5BMTc0MDMyMzI2OF5BMl5BanBnXkFtZTcwMzM2OTk1MQ@@._V1_Ratio1.0000_AL_.jpg)",
              "backgroundSize": "cover",
            }
          }
        />
        <div
          className="actorInfo"
        >
          <h3>
            Morgan Freeman
          </h3>
          <p>
            Ellis Boyd 'Red' Redding
          </p>
        </div>
      </div>
      <div
        className="actor"
      >
        <div
          className="imgContainer"
          style={
            Object {
              "backgroundImage": "url(https://m.media-amazon.com/images/M/MV5BMjUyZDQ0NjktZmM5ZS00NzcxLTliMWYtNWUxNDcyMmExZjU0XkEyXkFqcGdeQXVyMTE1MjA4NzM@._V1_Ratio1.0000_AL_.jpg)",
              "backgroundSize": "cover",
            }
          }
        />
        <div
          className="actorInfo"
        >
          <h3>
            Bob Gunton
          </h3>
          <p>
            Warden Norton
          </p>
        </div>
      </div>
      <div
        className="actor"
      >
        <div
          className="imgContainer"
          style={
            Object {
              "backgroundImage": "url(https://m.media-amazon.com/images/M/MV5BMTA1NjU3NDg1MTheQTJeQWpwZ15BbWU2MDI4OTcxMw@@._V1_Ratio1.0000_AL_.jpg)",
              "backgroundSize": "cover",
            }
          }
        />
        <div
          className="actorInfo"
        >
          <h3>
            William Sadler
          </h3>
          <p>
            Heywood
          </p>
        </div>
      </div>
      <div
        className="actor"
      >
        <div
          className="imgContainer"
          style={
            Object {
              "backgroundImage": "url(https://m.media-amazon.com/images/M/MV5BMTUxODY3NjAzMF5BMl5BanBnXkFtZTcwMTQ5MjYwNg@@._V1_Ratio1.0000_AL_.jpg)",
              "backgroundSize": "cover",
            }
          }
        />
        <div
          className="actorInfo"
        >
          <h3>
            Clancy Brown
          </h3>
          <p>
            Captain Hadley
          </p>
        </div>
      </div>
      <div
        className="actor"
      >
        <div
          className="imgContainer"
          style={
            Object {
              "backgroundImage": "url(https://m.media-amazon.com/images/M/MV5BMTgxMzc0MDAzNV5BMl5BanBnXkFtZTgwMzUzMTI0MzE@._V1_Ratio1.0000_AL_.jpg)",
              "backgroundSize": "cover",
            }
          }
        />
        <div
          className="actorInfo"
        >
          <h3>
            Gil Bellows
          </h3>
          <p>
            Tommy
          </p>
        </div>
      </div>
      <div
        className="actor"
      >
        <div
          className="imgContainer"
          style={
            Object {
              "backgroundImage": "url(https://m.media-amazon.com/images/M/MV5BODE5NDRlZTctMjkzNC00NGY3LThhYTItNzUxZjgzYzFmMjQxXkEyXkFqcGdeQXVyNjUxMjc1OTM@._V1_Ratio1.0000_AL_.jpg)",
              "backgroundSize": "cover",
            }
          }
        />
        <div
          className="actorInfo"
        >
          <h3>
            Mark Rolston
          </h3>
          <p>
            Bogs Diamond
          </p>
        </div>
      </div>
      <div
        className="actor"
      >
        <div
          className="imgContainer"
          style={
            Object {
              "backgroundImage": "url(https://m.media-amazon.com/images/M/MV5BMTg5MzkxMTkxOV5BMl5BanBnXkFtZTcwNTEzNTgxMw@@._V1_Ratio1.0000_AL_.jpg)",
              "backgroundSize": "cover",
            }
          }
        />
        <div
          className="actorInfo"
        >
          <h3>
            James Whitmore
          </h3>
          <p>
            Brooks Hatlen
          </p>
        </div>
      </div>
      <div
        className="actor"
      >
        <div
          className="imgContainer"
          style={
            Object {
              "backgroundImage": "url(https://m.media-amazon.com/images/M/MV5BMTQ0Mjc3NDA1OV5BMl5BanBnXkFtZTcwMTg3MDEyOA@@._V1_Ratio1.0000_AL_.jpg)",
              "backgroundSize": "cover",
            }
          }
        />
        <div
          className="actorInfo"
        >
          <h3>
            Jeffrey DeMunn
          </h3>
          <p>
            1946 D.A.
          </p>
        </div>
      </div>
      <div
        className="actor"
      >
        <div
          className="imgContainer"
          style={
            Object {
              "backgroundImage": "url(https://m.media-amazon.com/images/M/MV5BNDczZGZiZjMtZGRmMC00Y2I5LThiNzAtZGQzZWYzZWFiNzM0XkEyXkFqcGdeQXVyMjQwMDg0Ng@@._V1_Ratio1.0000_AL_.jpg)",
              "backgroundSize": "cover",
            }
          }
        />
        <div
          className="actorInfo"
        >
          <h3>
            Larry Brandenburg
          </h3>
          <p>
            Skeet
          </p>
        </div>
      </div>
      <div
        className="actor"
      >
        <div
          className="imgContainer"
          style={
            Object {
              "backgroundImage": "url(https://m.media-amazon.com/images/M/MV5BMjI0OTUxNjIyNF5BMl5BanBnXkFtZTcwNDE0MDcwOA@@._V1_Ratio1.0000_AL_.jpg)",
              "backgroundSize": "cover",
            }
          }
        />
        <div
          className="actorInfo"
        >
          <h3>
            Neil Giuntoli
          </h3>
          <p>
            Jigger
          </p>
        </div>
      </div>
      <div
        className="actor"
      >
        <div
          className="imgContainer"
          style={
            Object {
              "backgroundImage": "url(https://m.media-amazon.com/images/M/MV5BMjI2NDYwNzU0NV5BMl5BanBnXkFtZTcwMjYwMTcwOA@@._V1_Ratio1.0000_AL_.jpg)",
              "backgroundSize": "cover",
            }
          }
        />
        <div
          className="actorInfo"
        >
          <h3>
            Brian Libby
          </h3>
          <p>
            Floyd
          </p>
        </div>
      </div>
      <div
        className="actor"
      >
        <div
          className="imgContainer"
          style={
            Object {
              "backgroundImage": "url(https://m.media-amazon.com/images/M/MV5BMjE0MTU0NjU3Nl5BMl5BanBnXkFtZTYwNTgzNjY0._V1_Ratio1.0000_AL_.jpg)",
              "backgroundSize": "cover",
            }
          }
        />
        <div
          className="actorInfo"
        >
          <h3>
            David Proval
          </h3>
          <p>
            Snooze
          </p>
        </div>
      </div>
      <div
        className="actor"
      >
        <div
          className="imgContainer"
          style={
            Object {
              "backgroundImage": "url(https://m.media-amazon.com/images/M/MV5BMTQyNzMyNTUwNV5BMl5BanBnXkFtZTcwNjQ2MTcwOA@@._V1_Ratio1.0000_AL_.jpg)",
              "backgroundSize": "cover",
            }
          }
        />
        <div
          className="actorInfo"
        >
          <h3>
            Joseph Ragno
          </h3>
          <p>
            Ernie
          </p>
        </div>
      </div>
      <div
        className="actor"
      >
        <div
          className="imgContainer"
          style={
            Object {
              "backgroundImage": "url(https://m.media-amazon.com/images/M/MV5BMjZmOTU1ZDktNDdkZi00YjkxLWFiODYtMjUzZWVhYmFiYjIxXkEyXkFqcGdeQXVyMTI3MzAzOTM@._V1_Ratio1.0000_AL_.jpg)",
              "backgroundSize": "cover",
            }
          }
        />
        <div
          className="actorInfo"
        >
          <h3>
            Jude Ciccolella
          </h3>
          <p>
            Guard Mert
          </p>
        </div>
      </div>
      <div
        className="actor"
      >
        <div
          className="imgContainer"
          style={
            Object {
              "backgroundImage": "url(https://m.media-amazon.com/images/M/MV5BMTI2Mjc1NzExN15BMl5BanBnXkFtZTYwNjUzMzQz._V1_Ratio1.0000_AL_.jpg)",
              "backgroundSize": "cover",
            }
          }
        />
        <div
          className="actorInfo"
        >
          <h3>
            Paul McCrane
          </h3>
          <p>
            Guard Trout
          </p>
        </div>
      </div>
      <div
        className="actor"
      >
        <div
          className="imgContainer"
          style={
            Object {
              "backgroundImage": "url(https://imdb-api.com/images/original/nopicture.jpg)",
              "backgroundSize": "cover",
            }
          }
        />
        <div
          className="actorInfo"
        >
          <h3>
            Renee Blaine
          </h3>
          <p>
            Andy Dufresne's Wife
          </p>
        </div>
      </div>
      <div
        className="actor"
      >
        <div
          className="imgContainer"
          style={
            Object {
              "backgroundImage": "url(https://imdb-api.com/images/original/nopicture.jpg)",
              "backgroundSize": "cover",
            }
          }
        />
        <div
          className="actorInfo"
        >
          <h3>
            Scott Mann
          </h3>
          <p>
            Glenn Quentin
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
